nixos/modules/services/web-apps/coder.nix

```diff
@@ -0,0 +1,165 @@
+{ config, lib, options, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.coder;
+in {
+  options = {
+    services.coder = {
+      enable = mkEnableOption (lib.mdDoc "Coder enable the service");
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.coder;
+        description = lib.mdDoc ''
+          Package to use for the service
+        '';
+        defaultText = literalExpression "pkgs.coder";
+      };
+
+      homeDir = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          Home directory for coder user
+        '';
+        default = "/var/lib/coder";
+      };
+
+      virtualHost = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = lib.mdDoc ''
+          Name of the nginx virtualhost to use and setup. If null, do not setup any virtualhost.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          Listen address
+        '';
+        default = "127.0.0.1:3000";
+      };
+
+      accessUrl = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          Access URL should be a external IP address or domain with DNS records pointing to Coder
+        '';
+        default = null;
+        example = "https://coder.example.com";
+      };
+
+      wildcardAccessUrl = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          If you are providing TLS certificates directly to the Coder server, you must use a single certificate for the root and wildcard domains
+        '';
+        default = null;
+        example = "*.coder.example.com";
+      };
+
+      postgresqlUrl = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          Coder uses a PostgreSQL database to store users, workspace metadata, and other deployment information
+        '';
+        default = null;
+        example = "postgresql://root:root@database/coder?sslmode=disable";
+      };
+
+      tlsCert = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          The path to the TLS certificate.
+        '';
+        default = null;
+      };
+
+      tlsKey = mkOption {
+        type = types.nullOr types.str;
+        description = lib.mdDoc ''
+          The path to the TLS key.
+        '';
+        default = null;
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.coder = let
+      # see https://github.com/coder/coder/issues/4731
+      pgCtl = pkgs.writeScript "pg_ctl" ''
+        #!${pkgs.bash}/bin/bash
+
+        exec ${pkgs.postgresql_14}/bin/pg_ctl "$@" -o '--unix_socket_directories=${cfg.homeDir}'
```

Comment:
That's weird. I saw the issue but I don't understand why we should do that. Postgres should get launched through this service. 

The `postresql` module has a socket in `/run/postgresql` by default. But regardless of that it's up to the user whether to us a socket or connect by IP.

Reply:
coder has embeeded postgres support. I will remove it because its soo uggly 😂